src/components/common/theme-switcher/__tests__/theme-switcher.test.tsx

```diff
@@ -0,0 +1,125 @@
+import { render, screen, waitFor } from '@testing-library/react';
+import type { UserEvent } from '@testing-library/user-event';
+import userEvent from '@testing-library/user-event';
+import { NextIntlClientProvider } from 'next-intl';
+import type { ReactNode } from 'react';
+import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest';
+
+import { themes } from '@/configs/themes';
+
+import ThemeSwitcher from '../theme-switcher';
+
+const mocked = vi.hoisted(() => ({
+  setThemeMocked: vi.fn(),
+  resolvedThemeMocked: 'light',
+}));
+
+vi.mock('next-themes', async (importOriginal) => {
+  // eslint-disable-next-line @typescript-eslint/consistent-type-imports
+  const mod = await importOriginal<typeof import('next-themes')>();
+  return {
+    ...mod,
+    useTheme: () => ({
+      setTheme: mocked.setThemeMocked,
+      resolvedTheme: mocked.resolvedThemeMocked,
+    }),
+  };
+});
+
+const Component = () => {
+  return (
+    <NextIntlClientProvider
+      locale="en"
+      messages={{
+        components: {
+          common: {
+            themeSwitcher: {
+              label: 'Toggle theme',
+              tooltip: {
+                light: 'Switch to light theme',
+                dark: 'Switch to dark theme',
+              },
+            },
+          },
+        },
+      }}
+    >
+      <ThemeSwitcher />
+    </NextIntlClientProvider>
+  );
+};
+
+describe('ThemeSwitcher', () => {
+  let checkbox: HTMLInputElement;
+  let user: UserEvent;
+  let rerender: (ui: ReactNode) => void = () => {};
```

Comment:
**suggestion**: simplify type `(ui: ReactNode) => void = () => {}` to `(ui: ReactNode) => void`